### ISSUE TITLE
Enable enchanced code diff for jupyter notebooks

### ISSUE THREAD

**Issue by jpsalado92** [desktop/desktop#12121]
A feature to enchance `.ipynb` diff visualizations would be incredible for researchers, as right now diffs over these files show a hard to follow `json` structure.
![Sin título](https://user-images.githubusercontent.com/26083931/116868459-1adb5200-ac0f-11eb-8d08-727e452dbaa7.png)

Basically, it would be quite handy to have an option which, when showing diffs for .ipynb files, hides/unhides cell meta information. 
By doing this only the "cell_type" and "source" fields should remain.

A more complete option alternative, would be to somehow render `.ipynb` files in a way pycharm does. (See attached image)
![Sin título](https://user-images.githubusercontent.com/26083931/116868322-e1a2e200-ac0e-11eb-9854-2abbbea22f6a.png)

**Comment by steveward** [comment]
Thanks for the issue @jpsalado92. Rendering Jupyter notebooks is something that has been previously discussed in https://github.com/desktop/desktop/issues/5279, but it would require quite a few changes to implement. I'll cross-reference this issue to that one.

Diffing Jupyter notebooks is tricky since line-based diffs don't work well with them. It looks like [nbdime](https://nbdime.readthedocs.io/en/latest/) would be the best solution for this. At this time we don't have plans to implement any type of custom diffing directly into the app, but if we hear enough feedback about this it may be something we consider again in the future. 

It does look like you can set up `nbdime` as a custom diff/merge driver in Git. I would recommend keeping an eye on https://github.com/desktop/desktop/issues/1765 where we are tracking feedback about launching custom diff/merge tools from GitHub Desktop.

**Comment by jpsalado92** [comment]
Thanks for your comments and your time @steveward. Will check those issues.
Best regards.

JPS

**Comment by bjudkewitz** [comment]
> At this time we don't have plans to implement any type of custom diffing directly into the app, but if we hear enough feedback about this it may be something we consider again in the future.

+1 on this issue, right now the use of github desktop for jupyter notebooks is really limited

**Comment by manvillej** [comment]
So Github added a new feature preview for Rich Jupyter Notebook Diffs
![FireShot Capture 054 - GitHub - github com](https://github.com/desktop/desktop/assets/10823980/4635ba45-2bee-41eb-b4e1-b16622bfebfd)

Is there any progress on adding Rich Jupyter Notebook Diffs for Github Desktop?

**Comment by flockonus** [comment]
My number one reason to use Github Desktop is the good diff visualization, everything else quite easy to do in the terminal.

A bit frustrated this feature request was closed with no explanation. I can imagine actually executing python from an electron interface brings all sorts of challenges, but would hope some basic viz of the code + markdown can be achieved in plain JS given the format of .ipynb is some very straightforward json.

Right now committing python notebook is challenging coming up with a commit message since it's quite messy figuring out actual changes

**Comment by lucafossen** [comment]
Most diff viewers provided by Microsoft (VSCode, github.com etc.) now support notebook visualization. Many competitors to Github Desktop also support this. It's surprising that Microsoft's flagship diff viewer still doesn't after nearly four years.
